tests/v2/validation/fleet/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,210 @@
+	err = verifyGitCommit(u.client, gitStatus.Commit, repoObject.ID)
+	require.NoError(u.T(), err)
+}
+
+func (u *UpgradeTestSuite) TestGitRepoForceUpdate() {
+	u.session = session.NewSession()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(u.client, u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	log.Info("Cloning Git Repo")
+	repoName := namegenerator.AppendRandomString("repo-name")
+	_, err = u.sshNode.ExecuteCommand(fmt.Sprintf("cd ~/ && git clone %s %s", fleet.ExampleRepo, repoName))
+	require.NoError(u.T(), err)
+
+	log.Info("Creating Fleet repo")
+	repoObject, err := createFleetGitRepo(u.client, u.sshNode, repoName, namespace.Name, u.clusterName, u.cluster.ID, u.fleetSecretName)
+	require.NoError(u.T(), err)
+
+	log.Info("Getting GitRepo")
+	lastRepoObject, err := u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(repoObject.ID)
+	require.NoError(u.T(), err)
+
+	gitRepo := &v1alpha1.GitRepo{}
+	err = steveV1.ConvertToK8sType(lastRepoObject, gitRepo)
+	require.NoError(u.T(), err)
+
+	u.T().Log("Updating Fleet Repo")
+	_, err = u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).Update(lastRepoObject, gitRepo)
+	require.NoError(u.T(), err)
+
+	u.T().Log("Verifying the Fleet GitRepo")
+	err = fleet.VerifyGitRepo(u.client, repoObject.ID, u.cluster.ID, fmt.Sprintf("%s/%s", fleet.Namespace, u.clusterName))
+	require.NoError(u.T(), err)
+}
+
+func (u *UpgradeTestSuite) TestPauseFleetRepo() {
+	u.session = session.NewSession()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(u.client, u.cluster.ID)
+	require.NoError(u.T(), err)
+
+	log.Info("Cloning Git Repo")
+	repoName := namegenerator.AppendRandomString("repo-name")
+	_, err = u.sshNode.ExecuteCommand(fmt.Sprintf("cd ~/ && git clone %s %s", fleet.ExampleRepo, repoName))
+	require.NoError(u.T(), err)
+
+	log.Info("Creating Fleet repo")
+	repoObject, err := createFleetGitRepo(u.client, u.sshNode, repoName, namespace.Name, u.clusterName, u.cluster.ID, u.fleetSecretName)
+	require.NoError(u.T(), err)
+
+	log.Info("Getting GitRepo")
+	lastRepoObject, err := u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(repoObject.ID)
+	require.NoError(u.T(), err)
+
+	gitRepo := &v1alpha1.GitRepo{}
+	err = steveV1.ConvertToK8sType(lastRepoObject, gitRepo)
+	require.NoError(u.T(), err)
+
+	u.T().Log("Pausing Fleet Repo")
+	gitRepo.Spec.Paused = true
+	repoObject, err = u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).Update(lastRepoObject, gitRepo)
+	require.NoError(u.T(), err)
+
+	err = gitPushCommit(u.client, u.sshNode, repoName)
+	require.NoError(u.T(), err)
+
+	log.Info("Getting last GitRepo")
+	lastRepoObject, err = u.client.Steve.SteveType(extensionsfleet.FleetGitRepoResourceType).ByID(repoObject.ID)
+	require.NoError(u.T(), err)
+
+	log.Info("Checking last GitRepo was not updated")
```

Comment:
this isn't verifying that the gitRepo wasn't updated, but only checks that the gitRepoObject has `paused` set. Can you add a verify that repoObject.Commit == lastRepoObject.Commit? (whatever the actual objects look like)